spec/src/main/asciidoc/interoperability.asciidoc

```diff
@@ -419,6 +419,16 @@ public class RolesEndpoint {
 
 ```
 
+If there is no JWT in the request, an empty `JsonWebToken` is injected, which means all method calls to this token
+return `null`. Note that MP JWT will still perform Authentication and Authorization if the endpoint requires these
+verifications. Effectively, the injected empty token  is only visible on unauthenticated and unauthorized endpoints.
+
+If a JWT is sent to an endpoint that does not require Authentication and/or Authorization, the `JsonWebToken`
```

Comment:
@radcortez It looks good but I'd say it needs to be tweaked a bit, 
`If a JWT is sent to an endpoint that does not require Authentication and/or Authorization, its verification must still be performed before it is injected as `JsonWebToken`` or similar, given that `JsonWebToken` term is associated with an MP JWT interface

Reply:
Changed to `If a JWT is sent to an endpoint that does not require Authentication and/or Authorization, the `JsonWebToken`
verification must still be performed before being injected or retrieved from the `BeanManager`.`

Reply:
@radcortez IMHO there is no need to refer to Bean manager. Your original sentence looked good and simple, I've only asked to clarify that what should be validated is the token, `JsonWebToken` is an interface. We don't say for example `if a JsonWebToken is sent`...

Reply:
Ok, let me remove the last bit.

Reply:
@radcortez I'm sorry, but we start circling - may be it is my fault :-). All I'm asking for is that we don't say `JsonWebToken verification is still performed` because `JsonWebToken` is an interface. The implementations verify the raw tokens, not the interfaces, before making it accessible via `JsonWebToken`

Can we just say:
```
JWT must still be verified if it is sent to an endpoint that does not require Authentication and/or Authorization but has a `JsonWebToken` injection point.
```
?

Reply:
Sure, but that is why I've added the `BeanManager` portion in the first place. If I try ti retrieve and instance of `JsonWebToken` with the BeanManager, shouldn't that also be verified? The sentence seems to imply it only happens for injection points.

Reply:
@radcortez the phrase `JsonWebToken verification is still performed` does not make sense IMHO. Because `JsonWebToken` is a reserved word for the interface. So we don't verify the interfaces. And referring to some low level details like bean manager is also not needed. All I'm asking for is for a minor update to your original sentence, which will make it obvious that it is a token which still needs to be verified. Thanks

Reply:
@radcortez would it work for you if you just replace `JsonWebToken verification is still performed` with `JWT verification is still performed` given that use refer to a token as `JWT` in the same sentence ?

Reply:
No, that does not work - because it opens a new requirement that it is verified even if no injection point exists.
So perhaps the very 1st suggestion I made may work, whatever that can make it obvious without getting into the techicalities will do

Reply:
Yes, that is the issue that we need to figure out.

Does a token still gets validated if there is no injection point for `JsonWebToken`? What if I call CDI programatic API to retrieve the `JsonWebToken`. The one returned should also be valid.

That is why I believe that if we sent a token to an unprotected endpoint, it must always be verified. So there are two outcomes:

Unprotected resources
- IsTokenPresent -> validate, can warn in logs, but not fail the request, proceed to the endpoint
- TokenIsNotPresent -> proceed to the endpoint 

What do you think?

Reply:
@radcortez 

> Does a token still gets validated if there is no injection point for JsonWebToken

IMHO this is definitely a standalone separate issue. Lets complete this issue first. Thanks

Reply:
Ok, if we want to handle this is a separate issue, than I propose to drop the (or possible changes being discussed in the sentence):

> If a JWT is sent to an endpoint that does not require Authentication and/or Authorization, the `JsonWebToken` verification must still be performed.

And discuss and clarify the behaviour for the raised concern.

Reply:
@radcortez I disagree. The original issue has been about a case where a user has `JsonWebToken` injection point and would like to use this injection point. Therefore an issue about what to do when such injection point does not even exist is completely orthogonal. I'd like to have a sentence that makes it clear that when a `JsonWebToken` injection point exists the token if available must be verified.

Reply:
Yes, but we cannot just make the decision about validating if the injection point exists or not. I can go can call the CDI API programatically to retrieve an instance of the `JsonWebToken`. Shouldn't that be validated as well even if there is no injection point?

Reply:
@radcortez I'm not sure why are we are getting these lower-level technicalities, but either way, if the service code attempts to access a token via `JsonWebToken` `interface`, in whatever way they prefer, the verification of the token must happen. The minor update I suggested originally,
```
If a JWT is sent to an endpoint that does not require Authentication and/or Authorization then its verification must still be performed before it is injected as JsonWebToken``
```
had made it clear. If you think that CDI.getInstance  or whatever is not covered by this then

```
If a JWT is sent to an endpoint that does not require Authentication and/or Authorization then it still must be verified before it can be accessed via JsonWebToken interface`
```

certainly does IMHO

Reply:
Sorry, I guess we got a little bit lost through the discussion (at least me :) )

Yes. The second sentence should clear that concern. Thank you.